File: search/binary-search.js
// Binary search

const binarySearch = (arr, element) => {
    let first = 0
    let last = arr.length - 1
    let positions = -1
    let found = false
    let middle

        while (found === false && first <= last) {
            middle = Math.floor((first + last) / 2)
            if (arr[middle] === element) {
                found = true
                positions = middle
            } else if (arr[middle] > element) {
                last = middle + 1
            } else {
                first = middle - 1
            }
        }
        return positions
}

console.log(binarySearch([1,2,3],1))

File: data-structures/leetcode/array/BinarySearch.cs
namespace data_structures.leetcode.array;

public static class BinarySearch
{
    public static int SearchElement(int[] arr, int target)
    {
        if (arr.Length == 0) return -1;
       
        
        int first = 0;
        int last = arr.Length - 1;

        while (first <= last)
        {
            int middle = (last + first) / 2;

            if (arr[middle] > target)
            {
                last = middle - 1;
            }
            else if (arr[middle] < target)
            {
                first = middle + 1;
            }
            else
            {
                return middle;
            }
        }

        return -1;
    }
}